src/lib/InstantSearch.js

```diff
@@ -109,14 +120,25 @@ Usage: instantsearch({
       this.searchParameters
     );
 
+    if (this._searchFunction) {
+      this._originalHelperSearch = helper.search.bind(helper);
+      helper.search = this._wrappedSearch.bind(this);
```

Comment:
We might want to clone the helper first, to leave the original one "untouched". What do you think? cc @bobylito